xcube_stac/store.py

```diff
@@ -97,35 +117,92 @@ def get_item_collection(
         using the supplied *open_params*.
 
         Returns:
-            A tuple of the item collection containing all items
-            identified by *open_params* and data IDs corresponding to items
+            items: item collection containing all items identified by *open_params*
+            item_data_ids: data IDs corresponding to items
         """
-        return self.stac.get_item_collection(**open_params)
+        if self._searchable:
+            # not used
+            open_params.pop("variable_names", None)
+            # rewrite to "datetime"
+            time_range = open_params.pop("time_range", None)
+            if time_range:
+                open_params["datetime"] = "/".join(time_range)
+            items = self._catalog.search(**open_params).item_collection()
+        else:
+            items = self._get_items_nonsearchable_catalog(
+                self._catalog,
+                **open_params
+            )
+            items = pystac.ItemCollection(items)
+        item_data_ids = self.list_item_data_ids(items)
+        return items, item_data_ids
+
+    def get_item_data_id(self, item: pystac.Item) -> str:
+        """Generates the data ID of an item, which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            item: item/feature
+
+        Returns:
+            data ID of an item
+        """
+        id_parts = []
+        parent_item = item
+        while parent_item.STAC_OBJECT_TYPE != "Catalog":
+            id_parts.append(parent_item.id)
+            parent_item = parent_item.get_parent()
+        id_parts.reverse()
+        return self._data_id_delimiter.join(id_parts)
+
+    def get_item_data_ids(self, items: Iterable[pystac.Item]) -> Iterator[str]:
+        """Generates the data IDs of an item collection,
+        which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            items: item collection
+
+        Yields:
+            data ID of an item
+        """
+        for item in items:
+            yield self.get_item_data_id(item)
+
+    def list_item_data_ids(self, items: Iterable[pystac.Item]) -> List[str]:
+        """Generates a list of data IDs for a given item collection,
+        which follows the structure:
+
+            `collection_id_0/../collection_id_n/item_id`
+
+        Args:
+            items: item collection
+
+        Returns:
+            list of data IDs for a given item collection
+        """
+        return list(self.get_item_data_ids(items))
 
     def get_data_ids(
         self,
-        data_type: DataTypeLike = None,
```

Comment:
You're implementing the abstract method from `DataStore` so you should retain the arguments from its type signature even if you don't use them -- otherwise callers who don't know that their `DataStore` is actually a `StacDataStore`, could pass a `data_type` argument and get an unpleasant surprise.

Reply:
I changed it back. Further the method will give a warning if `data_type` is set to a value other than the default, as it is not used so far.